inference-engine/src/mkldnn_plugin/nodes/gather_nd.cpp

```diff
@@ -0,0 +1,229 @@
+
+        _batchNum = 1lu;
+        for (size_t i = 0; i < _batchDims; i++) {
+            _batchNum *= indicesDims[i];
+        }
+
+        _sliceRank = indicesDims[indicesDims.size() - 1];
+        _dataRank = dataDims.size() - _batchDims;
+        if (_sliceRank > _dataRank)
+            THROW_IE_EXCEPTION << _errorPrefix << " has invalid inputs shapes.";
+
+        _blockSize = 1;
+        for (size_t i = _sliceRank + _batchDims; i < dataDims.size(); i++) {
+            _blockSize *= dataDims[i];
+        }
+        _batchStep = 1;
+        for (size_t i = _batchDims; i < dataDims.size(); i++) {
+            _batchStep *= dataDims[i];
+        }
+
+        LayerConfig config;
+        DataConfig dataConfig, indicesConfig, outConfig;
+        dataConfig.desc = TensorDesc(dataPrecision, dataDims,
+            data->getTensorDesc().getLayoutByDims(dataDims));
+        config.inConfs.push_back(dataConfig);
+        indicesConfig.desc = TensorDesc(Precision::I32, indicesDims,
+            indices->getTensorDesc().getLayoutByDims(indicesDims));
+        config.inConfs.push_back(indicesConfig);
+
+        const auto& outDims = layer->outData[0]->getTensorDesc().getDims();
+        outConfig.desc = TensorDesc(dataPrecision, outDims,
+                layer->outData[0]->getTensorDesc().getLayoutByDims(outDims));
+        config.outConfs.push_back(outConfig);
+        config.dynBatchSupport = false;
+
+        confs.push_back(config);
+    }
+
+    StatusCode execute(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs, ResponseDesc *resp) noexcept override {
+        if (_blockSize > 1) {
+            gatherBlocks(inputs, outputs, resp);
+        } else {
+            switch (_dataTypeSize) {
+                case sizeof(PrecisionTrait<Precision::I32>::value_type):
+                    gatherElementwise<PrecisionTrait<Precision::I32>::value_type>(inputs, outputs, resp);
+                    break;
+                case sizeof(PrecisionTrait<Precision::I16>::value_type):
+                    gatherElementwise<PrecisionTrait<Precision::I16>::value_type>(inputs, outputs, resp);
+                    break;
+                case sizeof(PrecisionTrait<Precision::I8>::value_type):
+                    gatherElementwise<PrecisionTrait<Precision::I8>::value_type>(inputs, outputs, resp);
+                    break;
+                default:
+                    std::string errMsg = _errorPrefix + " has data input with unsupported precision: " +
+                        inputs[_dataIndex]->getTensorDesc().getPrecision().name();
+                    errMsg.copy(resp->msg, sizeof(resp->msg) - 1);
+                    return GENERAL_ERROR;
+            }
+        }
+
+        return OK;
+    }
+
+protected:
+    template <typename dataType>
+    void gatherElementwise(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs, ResponseDesc *resp) noexcept {
+        const dataType* data = inputs[_dataIndex]->cbuffer().as<const dataType*>() +
```

Comment:
nit: to my mind it will be a liitle bit easy to read in case aligning the name with dstData: `data` -> `srcData`. This also true for other 'data' variables like: `shiftedData`

Reply:
Renamed